backend/api/users.py

```diff
@@ -112,18 +113,24 @@ def update_user():
 
         if 'bio' in user:
             res.bio = user['bio']
+
         if 'roles' in user:
             if 'admin' in user['roles'] and not current_user.has_role('admin'):
                 raise ApiException("Current user has no permission to assign admin role.", 403)
-            res.roles = [g.user_datastore.find_or_create_role(role) for role in user['roles']]
+            if 'admin' not in user['roles'] and res.has_role('admin'):
+                for usr in g.user_datastore.user_model.objects:
+                    if usr.has_role('admin') and usr['email'] != user['email']:
+                        res.roles = [g.user_datastore.find_or_create_role(role)
+                                     for role in user['roles']]
```

Comment:
For loops do support else statements in python. I would like a nice error message in that case then, instead of a silent discard of the requested change.